rest/client/timeout_sender_test.go

```diff
@@ -73,7 +74,20 @@ func (s *logSenderSuite) SetupSuite() {
 }
 
 func (s *logSenderSuite) TearDownSuite() {
-	s.Require().NoError(os.RemoveAll(s.tempDir))
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	for {
+		if ctx.Err() != nil {
+			grip.Errorf("timed out when removing temp dir %s", s.tempDir)
+			return
+		}
+		err := os.RemoveAll(s.tempDir)
+		if err == nil {
+			return
+		}
+		grip.Error(errors.Wrapf(err, "failed to remove temp dir in %s, trying again", s.T().Name()))
+		time.Sleep(500 * time.Millisecond)
```

Comment:
I'd probably use a timer rather than sleeps, so youcan put the code in a select loop.

Reply:
There is a difference though - if the os.RemoveAll call takes a significant part of the timer interval, we'll retry "too soon," though of course too soon is defined pretty arbitrarily. I'd prefer to keep it as is

Reply:
oh jk I think this is what you meant